### PR TITLE
uqtk: add pyqutk variant to uqtk

### PR DESCRIPTION
**Pull request by padamson**
Addresses #25297

### CODE REVIEW

**Comment by padamson** [comment]
- [x] add py-pymc3 as a dependency.

File: var/spack/repos/builtin/packages/uqtk/package.py [hunk 1]
```diff
@@ -38,8 +48,21 @@ def cmake_args(self):
...
-        return [
+        args = [
             '-DCMAKE_SUNDIALS_DIR={0}'.format(spec['sundials'].prefix),
             '-DLAPACK_LIBRARIES={0}'.format(lapack_libs),
-            '-DBLAS_LIBRARIES={0}'.format(blas_libs)
+            '-DBLAS_LIBRARIES={0}'.format(blas_libs),
         ]
+
+        if self.spec.variants['pyuqtk'].value:
+            args.extend(['-DPyUQTk=ON'])
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/uqtk/package.py]: You can rewrite all of the above as:
```python
args = [
    self.define('CMAKE_SUNDIALS_DIR', spec['sundials'].prefix),
    self.define('LAPACK_LIBRARIES', lapack_libs),
    self.define('BLAS_LIBRARIES', blas_libs),
    self.define_from_variant('PyUQTk', 'pyuqtk')
]
```

File: var/spack/repos/builtin/packages/uqtk/package.py [hunk 2]
```diff
@@ -38,8 +48,21 @@ def cmake_args(self):
...
+
+        return args
+
+    def setup_run_environment(self, env):
+        if self.spec.variants['pyuqtk'].value:
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/uqtk/package.py]:
```suggestion
        if '+pyuqtk' in self.spec:
```

**Review by alalazo — CHANGES_REQUESTED** [review]
I left a few comments below. Let me know what you think.

File: var/spack/repos/builtin/packages/uqtk/package.py (hunk 1 above)
Comment (padamson) [reply var/spack/repos/builtin/packages/uqtk/package.py]: Yes, much better, thanks. I wasn't sure if this would work since UQTk expects `-DPyUQTk=ON`.

**Review by alalazo — APPROVED** [review]

**Pull request merged**